sklearn/svm/_libsvm.pyx

```diff
@@ -350,7 +361,8 @@ def predict(np.ndarray[np.float64_t, ndim=2, mode='c'] X,
                        class_weight_label.data, class_weight.data)
     model = set_model(&param, <int> nSV.shape[0], SV.data, SV.shape,
                       support.data, support.shape, sv_coef.strides,
-                      sv_coef.data, intercept.data, nSV.data, probA.data, probB.data)
+                      sv_coef.data, intercept.data, nSV.data, probA.data,
+                      probB.data, n_iter.data)
```

Comment:
You are right @adrinjalali. I think we can remove the `n_iter` parameter from `set_model` and set `model->n_iter = NULL` inside to avoid problems while freeing the auxiliary models created.